### PR TITLE
Review .gitignore changes

### PR DESCRIPTION
**Pull request by smilinjake**

### CODE REVIEW

**Review by psohn12 — APPROVED** [review]

**Comment by psohn12** [comment]
JUST KIDDING, STILL NEEDS REVIEW

File: client/src/components/ratings_reviews/RatingsAndReviews.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React, {component, useState, useEffect} from 'react';
```
Comment (aywgit) [inline client/src/components/ratings_reviews/RatingsAndReviews.js]: Purely out of curiosity: where and for what reason are you using the 'component' method that you are importing?

File: client/src/components/ratings_reviews/RatingsAndReviews.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+const RatingsAndReviews = () => {
+  // these are the states for the url and review components
+  const [reviewData, setReviewData] = useState([]);
+  const [pageNumber, setPageNumber] = useState(1);
+  const [count, setCount] = useState(2);
+  const [sort, setSort] = useState('relevant');
+  const url = 'https://app-hrsei-api.herokuapp.com/api/fec2/hratx/reviews/';
+  // states are needed for the sidebar component
+  const [metaData, setMetaData] = useState({});
```
Comment (aywgit) [inline client/src/components/ratings_reviews/RatingsAndReviews.js]: If we are using Redux, we would not use the useState() hook. We would pass the props in on line 11, via the useDispatch and useSelector hooks available in Redux:  https://thoughtbot.com/blog/using-redux-with-react-hooks

File: client/src/components/ratings_reviews/RatingsAndReviews.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        setMetaData(response.data);
+      })
+      .catch(() => console.log('there has been an error fetching the metadata for the sidebar component'));
+  };
+
+  // this is similar to component did mount because this will run after the initial rendering of the page.
+  useEffect(() => {
+    getMetaData();
+    getReviews();
+  }, []);
```
Comment (aywgit) [inline client/src/components/ratings_reviews/RatingsAndReviews.js]: This is definitely a hook we should be using!

File: client/src/components/ratings_reviews/RatingsAndReviews.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  const [pageNumber, setPageNumber] = useState(1);
+  const [count, setCount] = useState(2);
+  const [sort, setSort] = useState('relevant');
+  const url = 'https://app-hrsei-api.herokuapp.com/api/fec2/hratx/reviews/';
+  // states are needed for the sidebar component
+  const [metaData, setMetaData] = useState({});
+
+  // getting reviews for a product
+  const getReviews = () => {
+    axios.get(`${url}?page=${pageNumber}&count=${count}&sort=${sort}&product_id=${exampleData.id}`, { 'headers': { 'Authorization': `${GITHUB_API_KEY}`}})
```
Comment (aywgit) [inline client/src/components/ratings_reviews/RatingsAndReviews.js]: can replace exampleData.id with store.currentProduct.id, after importing the store

File: client/src/components/ratings_reviews/RatingsAndReviews.js (hunk 1 above)
Comment (smilinjake) [reply client/src/components/ratings_reviews/RatingsAndReviews.js]: honestly, i have not used component and i think we can get rid of it!

File: client/src/components/ratings_reviews/RatingsAndReviews.js (hunk 3 above)
Comment (smilinjake) [reply client/src/components/ratings_reviews/RatingsAndReviews.js]: its like the hook version of componentDidMount i believe

File: client/src/components/ratings_reviews/RatingsAndReviews.js (hunk 4 above)
Comment (smilinjake) [reply client/src/components/ratings_reviews/RatingsAndReviews.js]: definitely, ill do that tomorrow. Currently, it is set up to use just the example data, but ill set it up to use store data.

File: client/src/components/ratings_reviews/RatingsAndReviews.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    getReviews();
+  }, [sort]);
+
+  // if the data doesnt exist yet render null
+  return (
+    <div>
+      <div>{ metaData.ratings ? <SideBar metaData={metaData} /> : null }</div>
+      <div>Review count: {reviewData.length}</div>
+      <div>Sorted by:
+        <select onChange={() => {
```
Comment (psohn12) [inline client/src/components/ratings_reviews/RatingsAndReviews.js:68]: function here should have 'event' as argument (but you got your sort working without it? interesting)

File: client/src/components/ratings_reviews/SideBar.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+// this component needs to render:
+//  - the Average star rating for a clicked item
+//  - the percent of reviewers that recommend this product
+//  - the breakdown of star ratings 1 -5 star ratings
+//  - how the product fits on a scale from too small to too large
+//  - how comfortable the product is from poor to perfect
+const SideBar = (props) => {
+  // console.log('this is props from sidebar:', props.metaData);
+  // need to find the average star rating
+  const starRating = props.metaData.ratings;
```
Comment (psohn12) [inline client/src/components/ratings_reviews/SideBar.js:12]: average rating is calculated in overview too - maybe worth tossing it into store

**Review by psohn12 — APPROVED** [review]
no outstanding problems atm, but remember what doesn't work (as discussed) and fix it before pushing to dev

**Pull request merged**